src/MarginTrading.SettingsService.SqlRepositories/Repositories/AssetPairsRepository.cs

```diff
@@ -0,0 +1,167 @@
+                                                 "[Id] [nvarchar] (64) NOT NULL, " +
+                                                 "[Name] [nvarchar] (64) NOT NULL, " +
+                                                 "[BaseAssetId] [nvarchar] (64) NULL, " +
+                                                 "[QuoteAssetId] [nvarchar] (64) NULL, " +
+                                                 "[Accuracy] [int] NOT NULL, " +
+                                                 "[MarketId] [nvarchar] (64) NULL, " +
+                                                 "[LegalEntity] [nvarchar] (64) NULL, " +
+                                                 "[BasePairId] [nvarchar] (64) NULL, " +
+                                                 "[MatchingEngineMode] [nvarchar] (64) NULL, " +
+                                                 "[StpMultiplierMarkupBid] decimal (24,10) NULL, " +
+                                                 "[StpMultiplierMarkupAsk] decimal (24,10) NULL, " +
+                                                 "CONSTRAINT Id UNIQUE(Id)" +
+                                                 ");";
+        
+        private static Type DataType => typeof(IAssetPair);
+        private static readonly string GetColumns = "[" + string.Join("],[", DataType.GetProperties().Select(x => x.Name)) + "]";
+        private static readonly string GetFields = string.Join(",", DataType.GetProperties().Select(x => "@" + x.Name));
+        private static readonly string GetUpdateClause = string.Join(",",
+            DataType.GetProperties().Select(x => "[" + x.Name + "]=@" + x.Name));
+
+        private readonly IConvertService _convertService;
+        private readonly string _connectionString;
+        private readonly ILog _log;
+        
+        public AssetPairsRepository(IConvertService convertService, string connectionString, ILog log)
+        {
+            _convertService = convertService;
+            _log = log;
+            _connectionString = connectionString;
+            
+            using (var conn = new SqlConnection(_connectionString))
+            {
+                try { conn.CreateTableIfDoesntExists(CreateTableScript, TableName); }
+                catch (Exception ex)
+                {
+                    _log?.WriteErrorAsync(nameof(AssetPairsRepository), "CreateTableIfDoesntExists", null, ex);
+                    throw;
+                }
+            }
+        }
+
+        public async Task<bool> TryInsertAsync(IAssetPair obj)
+        {
+            using (var conn = new SqlConnection(_connectionString))
+            {
+                try
+                {
+                    await conn.ExecuteAsync(
+                        $"insert into {TableName} ({GetColumns}) values ({GetFields})",
+                        _convertService.Convert<IAssetPair, AssetPairEntity>(obj));
+                }
+                catch
```

Comment:
I think it is better not to swallow exceptions, at least let's add warning into log?

Reply:
Ok